packages/apollo-language-server/src/utilities/debug.ts

```diff
@@ -0,0 +1,60 @@
+import { IConnection } from "vscode-languageserver";
+
+// for errors (and other logs in debug mode) we want to print a stack trace showing where they were thrown.
+// This uses an Error's stack trace, removes the three frames regarding this file (since they're useless) and
+// returns the rest of the trace.
+const createAndTrimStackTrace = () => {
+  let stack: string | undefined = new Error().stack;
+  // remove the lines in the stack from _this_ function and the caller (in this file) and shorten the trace
+  return stack && stack.split("\n").length > 2
+    ? stack
+        .split("\n")
+        .slice(3, 7)
+        .join("\n")
+    : stack;
+};
+
+export class Debug {
+  public static connection?: IConnection;
+
+  public static SetConnection(conn: IConnection) {
```

Comment:
using a static `connection` in this class, so we don't need to instantiate the class and pass it around anywhere. The user can just import the class itself from this file and call it without worrying like in the `workspace` file below